packages/office-addin-usage-data/src/usageData.ts

```diff
@@ -94,11 +94,13 @@ export class OfficeAddinUsageData {
         this.usageDataOptIn();
       }
 
-      appInsights.setup(this.options.instrumentationKey)
-        .setAutoCollectExceptions(false)
-        .start();
-      this.usageDataClient = appInsights.defaultClient;
-      this.removeApplicationInsightsSensitiveInformation();
+      if (this.options.usageDataLevel === "on") {
+        appInsights.setup(this.options.instrumentationKey)
+          .setAutoCollectExceptions(false)
+          .start();
+        this.usageDataClient = appInsights.defaultClient;
```

Comment:
Please validate that when this is not turned on and this.usageDataClient is undefined, this is handled OK and doesn't lead to an exception.

Can we add test cases which provide validation for when the option is both on and off?

Reply:
We should add tests for when this is turned off. I think most if not all tests only have usageDataLevel set to on

Reply:
We have a couple tests that set usageDataLevel to off and then call reportEvent and reportError and ensure that the event and exception static count hasn't changed. These tests pass with with this change:

  describe("Test getEventsSent method", () => {
    it("should return amount of events successfully sent", () => {
      addInUsageData.setUsageDataOff();
      const testEvent = {
        Test1: [true, 100],
        ScriptType: ["Java", 1],
      };
      addInUsageData.reportEvent("office-addin-usage-data", testEvent);
      assert.equal(addInUsageData.getEventsSent(), 1);
    });
  });
  describe("Test getExceptionsSent method", () => {
    it("should return amount of exceptions successfully sent ", () => {
      addInUsageData.setUsageDataOff();
      addInUsageData.reportError("TestData", err);
      assert.equal(addInUsageData.getExceptionsSent(), 1);
    });
  });

We have tests for when usageDataLevel is set to on:

  describe("Test reportEvent method", () => {
    it("should track event of object passed in with a project name", () => {
      const testEvent = {
        Test1: [true, 100],
        ScriptType: ["JavaScript", 1],
      };
      addInUsageData.reportEvent("office-addin-usage-data", testEvent);
      assert.equal(addInUsageData.getEventsSent(), 1);
    });
  });
  describe("Test reportError method", () => {
    it("should send usage data exception", () => {
      addInUsageData.reportError("ReportErrorCheck", err);
      assert.equal(addInUsageData.getExceptionsSent(), 1);
    });
  });

Reply:
It might be nice to have more validation of things not happening when it is off. The tests seems sufficient in terms of making this fix.